alf/trainers/policy_trainer.py

```diff
@@ -663,18 +679,33 @@ def play(root_dir,
             buffer_size=num_episodes, reward_shape=env.reward_spec().shape),
         alf.metrics.AverageEpisodeLengthMetric(buffer_size=num_episodes),
     ]
+
     while episodes < num_episodes:
-        time_step, policy_state, trans_state, info = _step(
+
+        # transform time step
+        transformed_time_step, trans_state = algorithm.transform_timestep(
+            time_step, trans_state)
+
+        time_step, policy_step, trans_state = _step(
             algorithm=algorithm,
             env=env,
             time_step=time_step,
             policy_state=policy_state,
             trans_state=trans_state,
             epsilon_greedy=epsilon_greedy,
             metrics=metrics)
+        policy_state = policy_step.state
+
         episode_length += 1
+
+        is_last_step = time_step.is_last() or (episode_length >=
+                                               max_episode_length > 0)
+
         if recorder:
-            recorder.capture_frame(info)
+            recorder.capture_frame(
+                transformed_time_step._replace(step_type=StepType.LAST) \
+                        if is_last_step else transformed_time_step,
```

Comment:
Would be better to just pass in is_last_step as an argument to capture_frame instead of modifying the step_type (in a hacky way).

Reply:
Done